### DISCUSSION TITLE
Render client (browser) only components with suspense fallback on server

### DISCUSSION THREAD

**Discussion by alexandernanberg** [vercel/next.js#52027] [Ideas]
### Goals

1. Rendering server fallback for browser only components triggers the error overlay when following the [official docs](https://react.dev/reference/react/Suspense#providing-a-fallback-for-server-errors-and-server-only-content )

### Background

https://react.dev/reference/react/Suspense#providing-a-fallback-for-server-errors-and-server-only-content 

### Proposal

Either by Next providing a custom error class that won't trigger the error overlay. 

E.g.

```tsx
<Suspense fallback={<Loading />}>
  <Chat />
</Suspense>

function Chat() {
  if (typeof window === 'undefined') {
    throw ClientOnlyError('Chat should only render on the client.');
  }
  // ...
}
```

or by exposing a component that hides this completely for consumers

```tsx
<Suspense fallback={<Loading />}>
  <ClientOnly>
    <Chat />
  </ClientOnly>
</Suspense>
```

**Comment by Chanki-Min** [comment]
Same problem here. client-only component shoud throw error in server-side runtime. Next.js should give developer a workaround to do not display non-nessesary error overlay.
